driver-core/src/main/resources/META-INF/native-image/reflect-config.json

```diff
@@ -0,0 +1,248 @@
+},
+{
+  "name":"com.mongodb.crypt.capi.CAPI",
+  "allPublicFields":true,
+  "queryAllDeclaredMethods":true
+},
+{
+  "name":"com.mongodb.crypt.capi.CAPI$cstring",
+  "methods":[{"name":"<init>","parameterTypes":[] }]
+},
+{
+  "name":"com.mongodb.crypt.capi.CAPI$mongocrypt_binary_t",
+  "methods":[{"name":"<init>","parameterTypes":[] }]
+},
+{
+  "name":"com.mongodb.crypt.capi.CAPI$mongocrypt_crypto_fn",
+  "queryAllDeclaredMethods":true,
+  "queryAllPublicMethods":true
+},
+{
+  "name":"com.mongodb.crypt.capi.CAPI$mongocrypt_ctx_t",
+  "methods":[{"name":"<init>","parameterTypes":[] }]
+},
+{
+  "name":"com.mongodb.crypt.capi.CAPI$mongocrypt_hash_fn",
+  "queryAllDeclaredMethods":true,
+  "queryAllPublicMethods":true
+},
+{
+  "name":"com.mongodb.crypt.capi.CAPI$mongocrypt_hmac_fn",
+  "queryAllDeclaredMethods":true,
+  "queryAllPublicMethods":true
+},
+{
+  "name":"com.mongodb.crypt.capi.CAPI$mongocrypt_kms_ctx_t",
+  "methods":[{"name":"<init>","parameterTypes":[] }]
+},
+{
+  "name":"com.mongodb.crypt.capi.CAPI$mongocrypt_log_fn_t",
+  "queryAllDeclaredMethods":true,
+  "queryAllPublicMethods":true
+},
+{
+  "name":"com.mongodb.crypt.capi.CAPI$mongocrypt_random_fn",
+  "queryAllDeclaredMethods":true,
+  "queryAllPublicMethods":true
+},
+{
+  "name":"com.mongodb.crypt.capi.CAPI$mongocrypt_status_t",
+  "methods":[{"name":"<init>","parameterTypes":[] }]
+},
+{
+  "name":"com.mongodb.crypt.capi.CAPI$mongocrypt_t",
+  "methods":[{"name":"<init>","parameterTypes":[] }]
+},
+{
+  "name":"boolean",
+  "fields":[{"name":"OPTIONS"}, {"name":"STRING_ENCODING"}, {"name":"STRUCTURE_ALIGNMENT"}, {"name":"TYPE_MAPPER"}]
+},
+{
+  "name":"com.sun.crypto.provider.AESCipher$General",
+  "methods":[{"name":"<init>","parameterTypes":[] }]
+},
+{
+  "name":"com.sun.crypto.provider.HmacCore$HmacSHA256",
+  "methods":[{"name":"<init>","parameterTypes":[] }]
+},
+{
+  "name":"com.sun.crypto.provider.HmacCore$HmacSHA512",
+  "methods":[{"name":"<init>","parameterTypes":[] }]
+},
+{
+  "name":"com.sun.jna.CallbackProxy",
+  "methods":[{"name":"callback","parameterTypes":["java.lang.Object[]"] }]
+},
+{
+  "name":"com.sun.jna.Pointer",
+  "fields":[{"name":"OPTIONS"}, {"name":"STRING_ENCODING"}, {"name":"STRUCTURE_ALIGNMENT"}, {"name":"TYPE_MAPPER"}]
+},
+{
+  "name":"com.sun.jna.Structure$FFIType",
+  "allDeclaredFields":true,
+  "queryAllPublicConstructors":true,
+  "fields":[{"name":"OPTIONS"}, {"name":"STRING_ENCODING"}, {"name":"STRUCTURE_ALIGNMENT"}, {"name":"TYPE_MAPPER"}],
+  "methods":[{"name":"<init>","parameterTypes":[] }]
+},
+{
+  "name":"com.sun.jna.Structure$FFIType$size_t",
+  "methods":[{"name":"<init>","parameterTypes":[] }]
+},
+{
+  "name":"com.sun.jna.ptr.PointerByReference",
+  "fields":[{"name":"OPTIONS"}, {"name":"STRING_ENCODING"}, {"name":"STRUCTURE_ALIGNMENT"}, {"name":"TYPE_MAPPER"}],
+  "methods":[{"name":"<init>","parameterTypes":[] }]
+},
+{
```

Comment:
`STRUCTURE_ALIGNMENT` suggests that it may be, though I don't know for sure. Even if I delete this entry, as well as the entries for `void`, `long`, `int`, `NativeImageApp` still works.

Reply:
The reason behind
```
{
  "name":"com.sun.org.apache.xerces.internal.jaxp.SAXParserFactoryImpl",
  "methods":[{"name":"<init>","parameterTypes":[] }]
}
```
is the code in Logback:
```
ch.qos.logback.core.joran.GenericXMLConfigurator#doConfigure(org.xml.sax.InputSource)
├── ch.qos.logback.core.joran.GenericXMLConfigurator#populateSaxEventRecorder(org.xml.sax.InputSource)
│   └── ch.qos.logback.core.joran.event.SaxEventRecorder#recordEvents(org.xml.sax.InputSource)
│       └── ch.qos.logback.core.joran.event.SaxEventRecorder#buildSaxParser()
│           └── javax.xml.parsers.SAXParserFactory#newInstance()
│               └── javax.xml.parsers.FactoryFinder#find(java.lang.Class,java.lang.String)
│                   └── javax.xml.parsers.FactoryFinder#newInstance(java.lang.Class,java.lang.String,java.lang.ClassLoader,boolean)
│                       └── javax.xml.parsers.FactoryFinder#newInstance(java.lang.Class,java.lang.String,java.lang.ClassLoader,boolean,boolean)
│                           ├── java.lang.reflect.Constructor#newInstance(java.lang.Object[]) - [ {   "name":"com.sun.org.apache.xerces.internal.jaxp.SAXParserFactoryImpl",   "methods":[{"name":"<init>","parameterTypes":[] }] } ]
│                           ├── java.lang.Class#getConstructor(java.lang.Class[]) - [ {   "name":"com.sun.org.apache.xerces.internal.jaxp.SAXParserFactoryImpl",   "methods":[{"name":"<init>","parameterTypes":[] }] } ]
│                           └── javax.xml.parsers.FactoryFinder#getProviderClass(java.lang.String,java.lang.ClassLoader,boolean,boolean)
│                               └── java.lang.Class#forName(java.lang.String,boolean,java.lang.ClassLoader) - [ {   "name":"com.sun.org.apache.xerces.internal.jaxp.SAXParserFactoryImpl" } ]
```

Reply:
Looks like one of the reasons behind
```
{
  "name":"int",
  "fields":[{"name":"OPTIONS"}, {"name":"STRING_ENCODING"}, {"name":"STRUCTURE_ALIGNMENT"}, {"name":"TYPE_MAPPER"}]
}
```
is
```
com.sun.jna.Native#lookupField(java.lang.Class,java.lang.String,java.lang.Class)
└── java.lang.Class#getField(java.lang.String) - [ {   "name":"boolean",   "fields":[{"name":"STRING_ENCODING"}, {"name":"STRUCTURE_ALIGNMENT"}, {"name":"TYPE_MAPPER"}] }, {   "name":"com.sun.jna.Pointer",   "fields":[{"name":"STRING_ENCODING"}, {"name":"STRUCTURE_ALIGNMENT"}, {"name":"TYPE_MAPPER"}] }, {   "name":"com.sun.jna.ptr.PointerByReference",   "fields":[{"name":"STRING_ENCODING"}, {"name":"STRUCTURE_ALIGNMENT"}, {"name":"TYPE_MAPPER"}] }, {   "name":"int",   "fields":[{"name":"STRING_ENCODING"}, {"name":"STRUCTURE_ALIGNMENT"}, {"name":"TYPE_MAPPER"}] }, {   "name":"long",   "fields":[{"name":"STRING_ENCODING"}, {"name":"STRUCTURE_ALIGNMENT"}, {"name":"TYPE_MAPPER"}] }, {   "name":"void",   "fields":[{"name":"STRING_ENCODING"}, {"name":"STRUCTURE_ALIGNMENT"}, {"name":"TYPE_MAPPER"}] } ]
```